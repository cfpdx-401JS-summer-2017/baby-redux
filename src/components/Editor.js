import React from 'react';
import { updateSalutation, updateName, resetForm } from '../greeter.actions';

export function Editor({ salutation, name, dispatch }) {
  return (
    <div>
      <input
        type="text"
        placeholder="salutation"
        value={salutation}
        onChange={({ target }) =>
          dispatch(updateSalutation({ salutation: target.value, name: name }))}
      />
      <input
        type="text"
        placeholder="name"
        value={name}
        onChange={({ target }) =>
          dispatch(updateName({ salutation: salutation, name: target.value }))}
      />
      <button onClick={() => dispatch(resetForm())} />
    </div>
  );
}
