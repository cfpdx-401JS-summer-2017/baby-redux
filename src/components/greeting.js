
import React from 'react';
import { connect } from 'react-redux';
import { changeName, changeSalutation, reset } from '../greeting.actions';

export function Greeting({ name, salutation, dispatch }) {
    return (
        <div>
            <div>
                <span>{salutation} </span>
                <span>{name}</span>!
            </div>

            <div>
                <label> Salutation <input value={salutation} 
                    onChange={({ target }) => dispatch(changeSalutation(target.value))} /></label>
            </div>
            <div>
                <label> Name <input name="name" value={name}
                    onChange={({ target }) => dispatch(changeName(target.value))} /></label>
            </div>
            <div>
                <button onClick={() => dispatch(reset())}>Reset</button>
            </div>
        </div>
    )
}

export default connect(
    state => ({
        name: state.name,
        salutation: state.salutation
    })
)(Greeting);