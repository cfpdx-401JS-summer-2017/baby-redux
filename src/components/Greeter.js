import React from 'react';
import { connect } from 'react-redux';
import { Editor } from './Editor';

export function Greeter({salutation, name, dispatch}) {
  console.log(salutation, name);
  return (
    <div>
      <div>
        <span>{salutation}</span>
        <span>{name}</span>!
      </div>
      <Editor salutation={salutation} name={name} dispatch={dispatch}/>
    </div>
  );
}

export default connect(
  state => ({
  salutation: state.salutation,
  name: state.name
}))(Greeter);
