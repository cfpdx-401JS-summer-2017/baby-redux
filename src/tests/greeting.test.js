
import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import { Greeting } from '../components/greeting';

describe('Greeting', () => {

  it('displays empty input fields and reset button', () => {
    const wrapper = shallow(<Greeting />);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  it('displays "Hello John!" in proper span tags', () => {
    const wrapper = shallow(<Greeting name="John" salutation="Hello" />);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});