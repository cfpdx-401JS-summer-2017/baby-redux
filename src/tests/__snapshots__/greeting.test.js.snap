// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Greeting displays "Hello John!" in proper span tags 1`] = `
<div>
  <div>
    <span>
      Hello
       
    </span>
    <span>
      John
    </span>
    !
  </div>
  <div>
    <label>
       Salutation 
      <input
        onChange={[Function]}
        value="Hello"
      />
    </label>
  </div>
  <div>
    <label>
       Name 
      <input
        name="name"
        onChange={[Function]}
        value="John"
      />
    </label>
  </div>
  <div>
    <button
      onClick={[Function]}
    >
      Reset
    </button>
  </div>
</div>
`;

exports[`Greeting displays empty input fields and reset button 1`] = `
<div>
  <div>
    <span>
       
    </span>
    <span />
    !
  </div>
  <div>
    <label>
       Salutation 
      <input
        onChange={[Function]}
      />
    </label>
  </div>
  <div>
    <label>
       Name 
      <input
        name="name"
        onChange={[Function]}
      />
    </label>
  </div>
  <div>
    <button
      onClick={[Function]}
    >
      Reset
    </button>
  </div>
</div>
`;
